src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregationManager.cs

```diff
@@ -0,0 +1,274 @@
+                    InstrumentState? state = GetInstrumentState(instrument);
+                    if (state != null)
+                    {
+                        _beginInstrumentMeasurements(instrument);
+                        listener.EnableMeasurementEvents(instrument, state);
+                    }
+                },
+                MeasurementsCompleted = (instrument, cookie) =>
+                {
+                    _endInstrumentMeasurements(instrument);
+                    RemoveInstrumentState(instrument, (InstrumentState)cookie!);
+                }
+            };
+            _listener.SetMeasurementEventCallback<double>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+            _listener.SetMeasurementEventCallback<float>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+            _listener.SetMeasurementEventCallback<long>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+            _listener.SetMeasurementEventCallback<int>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+            _listener.SetMeasurementEventCallback<short>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+            _listener.SetMeasurementEventCallback<byte>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+            _listener.SetMeasurementEventCallback<decimal>((i, m, l, c) => ((InstrumentState)c!).Update((double)m, l));
+        }
+
+
+
+        public AggregationManager Include(string meterName)
+        {
+            Include(i => i.Meter.Name == meterName);
+            return this;
+        }
+
+        public AggregationManager Include(string meterName, string instrumentName)
+        {
+            Include(i => i.Meter.Name == meterName && i.Name == instrumentName);
+            return this;
+        }
+
+        public AggregationManager Include(Predicate<Instrument> instrumentFilter)
+        {
+            lock (this)
+            {
+                _instrumentConfigFuncs.Add(instrumentFilter);
+            }
+            return this;
+        }
+
+        public AggregationManager SetCollectionPeriod(TimeSpan collectionPeriod)
+        {
+            lock (this)
+            {
+                _collectionPeriod = collectionPeriod;
+            }
+            return this;
+        }
+
+        public void Start()
+        {
+            // if already started or already stopped we can't be started again
+            if (_collectTask != null || _cts.IsCancellationRequested)
+            {
+                // correct usage from internal code should never get here
+                Debug.Assert(false);
+                throw new InvalidOperationException("Start can only be called once");
+            }
+
+            if (_collectionPeriod.TotalSeconds < 1)
+            {
+                throw new InvalidOperationException("CollectionPeriod must be >= 1 sec");
+            }
+
+            var token = _cts.Token;
```

Comment:
```suggestion
            CancellationToken token = _cts.Token;
```
BTW: `token` is a quite generic name, can this be named `cancellationToken`?